From 6382d62287adc512b78a2c3c9338e3026bd472a5 Mon Sep 17 00:00:00 2001
From: Timothy Day <tday141@gmail.com>
Date: Sat, 9 May 2026 13:34:54 -0400
Subject: [PATCH 02/10] userxattr by default

---
 fs/overlayfs/params.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/overlayfs/params.c b/fs/overlayfs/params.c
index c93fcaa45d4a..56349c64e8c4 100644
--- a/fs/overlayfs/params.c
+++ b/fs/overlayfs/params.c
@@ -985,6 +985,10 @@ int ovl_fs_params_verify(const struct ovl_fs_context *ctx,
 	}
 
 
+	/* Auto-enable userxattr when mounting without CAP_SYS_ADMIN */
+	if (!config->userxattr && !capable(CAP_SYS_ADMIN))
+		config->userxattr = true;
+
 	/* Resolve userxattr -> !redirect && !metacopy dependency */
 	if (config->userxattr) {
 		if (set.redirect &&
-- 
2.43.0

